### PR TITLE
Log when metric records are processed, up to once per minute.

### PR DESCRIPTION
**Pull request by anwar6953**
Log when metric records are processed, up to once per minute.
The current logging has a problem that its possible that the records processing is never logged, if the number of total records processed is never divisible by 1000.

Something to consider: this can get logged up to 1440 times per day, which may be more verbose than necessary.

http://builds.cask.co/browse/CDAP-RUT313-1

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MetricsMessageCallback.java [hunk 1]
```diff
@@ -89,16 +91,16 @@ public void onReceived(Iterator<FetchedMessage> messages) {
...
 
-    recordProcessed += records.size();
-    if (recordProcessed % 1000 == 0) {
-      LOG.info("{} metrics records processed", recordProcessed);
-      LOG.info("Last record time: {}", records.get(records.size() - 1).getTimestamp());
+    recordsProcessed += records.size();
+    // avoid logging more than once a minute
+    if (System.currentTimeMillis() > lastLoggedMillis + TimeUnit.MINUTES.toMillis(1)) {
+      lastLoggedMillis = System.currentTimeMillis();
+      LOG.info("{} metrics records processed. Last record time: {}.",
```
Comment (anew) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MetricsMessageCallback.java:102]: To reduce verbosity, you could combine these tow messages as they are always logged together. 
Also, I would rephrase to: N metrics records processed in the last minute.

Comment (anwar6953) [reply cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MetricsMessageCallback.java:102]: Combined the log messages.
Its a running count, so I didn't rephrase it.

Comment (anew) [reply cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MetricsMessageCallback.java:102]: ah I see. LGTM

**Review by anew — APPROVED** [review]
